### PR TITLE
Feature/nr 220413: Issues with "codestream/codemarks/codemark displayed" event

### PR DESCRIPTION
**Pull request by reneelpetit**
https://new-relic.atlassian.net/browse/NR-220413
https://new-relic.atlassian.net/browse/NR-220932

### CODE REVIEW

File: shared/ui/Stream/SearchResult.tsx [hunk 1]
```diff
@@ -129,6 +130,17 @@ export default function SearchResult(props: Props) {
 	});
 
 	const selectResult = () => {
+		const session = store.getState().session;
```
Comment (bcanzanella) [inline shared/ui/Stream/SearchResult.tsx:133]: change to `const { users, session }= store.getState();` then access `users` and `session` from that.

File: shared/ui/Stream/SearchResult.tsx [hunk 2]
```diff
@@ -129,6 +130,17 @@ export default function SearchResult(props: Props) {
...
+		const users = store.getState().users;
+		const currentUser = session.userId || users[session.userId!];
```
Comment (bcanzanella) [inline shared/ui/Stream/SearchResult.tsx:135]: `users[session.userId!]` returns a userId or a userObject? if it's an object you'll need to get the id -- i suspect the `toString()` below was an attempt around that...

File: shared/ui/index.tsx [hunk 3]
```diff
@@ -417,18 +419,32 @@ function listenForEvents(store) {
...
 		if (Object.keys(codemarks).length === 0) {
 			await store.dispatch(fetchCodemarks());
 			codemarks = store.getState().codemarks;
 		}
 
+		const currentUser = session.userId || users[session.userId!];
+
 		const codemark = getCodemark(codemarks, e.codemarkId);
+		HostApi.instance.track("codestream/codemarks/codemark displayed", {
+			meta_data: `codemark_location: source_file`,
```
Comment (bcanzanella) [inline shared/ui/index.tsx:441]: make that `source_file` come through as an optional argument from the Notification

File: vscode/src/commands.ts [hunk 4]
```diff
@@ -506,13 +506,6 @@ export class Commands implements Disposable {
...
 
-		Container.agent.telemetry.track("codestream/codemarks/codemark displayed", {
-			meta_data: `codemark_location: source_file`,
-			meta_data_2: "codemark_type: comment",
-			meta_data_3: `following: false`,
-			event_type: "modal_display"
-		});
-
 		const { codemarkId: _codemarkId, ...options } = args;
 		return Container.sidebar.openCodemark(args.codemarkId, options);
```
Comment (bcanzanella) [inline vscode/src/commands.ts:510]: `Container.sidebar.openCodemark(args.codemarkId, {source: "source_file", ...options });`

File: shared/ui/index.tsx (hunk 3 above)
Comment (bcanzanella) [reply shared/ui/index.tsx:441]: (see below)

File: shared/ui/Stream/ActivityPanel.tsx [hunk 5]
```diff
@@ -541,7 +541,7 @@ export const ActivityPanel = () => {
 													: ""
 											}`,
 											meta_data_3: `following: ${(codemark.followerIds || []).includes(
-												derivedState.currentUserId || ""
+												derivedState.currentUserId!
```
Comment (bcanzanella) [inline shared/ui/Stream/ActivityPanel.tsx:544]: is currentUserId guaranteed here? i see the removal of the empty string fallback...

File: shared/ui/Stream/SearchResult.tsx (hunk 2 above)
Comment (reneelpetit) [reply shared/ui/Stream/SearchResult.tsx:135]: I only saw a userId, I didn't know it could also be a userObject. I'll change it, thanks

Comment (bcanzanella) [reply shared/ui/Stream/SearchResult.tsx:135]: let me clarify... `users[session.userId!]` returns an `object` not a string. you would need to do `users[session.userId!]?.id`

File: shared/ui/index.tsx (hunk 3 above)
Comment (bcanzanella) [reply shared/ui/index.tsx:441]: it does in the current use-case, but anyone can trigger a `ShowCodemarkNotificationType` notification, not just things that come from source code.

File: shared/ui/index.tsx [hunk 6]
```diff
@@ -140,9 +140,15 @@ import translations from "./translations/en";
 import { parseProtocol } from "./utilities/urls";
 import { HostApi } from "./webview-api";
 import { parseId } from "./utilities/newRelic";
+import { SessionState } from "./store/session/types";
+import { UsersState } from "./store/users/types";
 
 // import translationsEs from "./translations/es";
 
+export interface SourceOptions {
```
Comment (bcanzanella) [inline shared/ui/index.tsx:148]: nice!!!

File: shared/ui/index.tsx [hunk 7]
```diff
@@ -417,18 +423,34 @@ function listenForEvents(store) {
...
+			users: UsersState;
+			source: SourceOptions;
 		} = store.getState();
 
 		if (Object.keys(codemarks).length === 0) {
 			await store.dispatch(fetchCodemarks());
 			codemarks = store.getState().codemarks;
 		}
 
+		const currentUser = session.userId || users[session.userId!];
```
Comment (bcanzanella) [inline shared/ui/index.tsx:442]: this should be:

`const currentUser = session.userId || users[session.userId!].id`;

(not in my IDE, so please double check)

**Review by bcanzanella — CHANGES_REQUESTED** [review]
see one comment

File: shared/ui/index.tsx (hunk 7 above)
Comment (reneelpetit) [reply shared/ui/index.tsx:442]: oh weird, I swear I already did that ....

**Pull request merged**